src/openms/source/CHEMISTRY/ModificationsDB.cpp

```diff
@@ -571,8 +524,11 @@ namespace OpenMS
       {
         // the mod has so far not been mapped to a unimod mod
         // first check whether the mod is specific
-        if ( (it->second.getOrigin().size() == 1 && it->second.getOrigin() != "X") ||
-              (it->second.getOrigin() == "N-term") || (it->second.getOrigin() == "C-term"))
+        // @TODO: allow terminal specificity (currently breaks various tests,
```

Comment:
No, it was broken before. There were tests for "origin = C-term" and "origin = N-term" in the condition, but those values were never set (except for Eugen's handling of cross-links).